cocotb/handle.py

```diff
@@ -530,11 +519,69 @@ def _range_iter(self, left, right):
-        return result
+        """Get value of array by iterating through sub-handles in left-to-right order.
+
+        Arrays with ascending ranges return list with values from sub-handles mapped thusly:
+            Verilog [4:7]       => [4, 5, 6, 7]
+            VHDL    (4 to 7)    => [4, 5, 6, 7]
+
+        Arrays with descending ranges:
+            Verilog [7:4]           => [7, 6, 5, 4]
+            VHDL    (7 downto 4)    => [7, 6, 5, 4]
+
+        Given an HDL array, the returned list contains the values of each sub-handle
+        in the order indicated in the table:
+        +----------------+-----------------------+------------------+
+        | Verilog        | VHDL                  | Python list      |
+        +================+=======================+==================+
+        | ``array[4:7]`` | ``array(4 to 7)``     | ``[4, 5, 6, 7]`` |
+        +----------------+-----------------------+------------------+
+        | ``array[7:4]`` | ``array(7 downto 4)`` | ``[7, 6, 5, 4]`` |
+        +----------------+-----------------------+------------------+
+        """
+        # Don't use self.__iter__, because it has an unwanted `except IndexError`
+        return [
+            self[i]
+            for i in self._range_iter(self._range[0], self._range[1])
+        ]
+
+    @value.setter
+    def value(self, value):
+        """Assign value from a list of same length to an array in left-to-right order.
+        Index 0 of the list maps to the left-most index in the array.
+
+        Given the following array with an ascending range:
+            Verilog [4:7]
+            VHDL    (4 to 7)
+        Array index 4 gets value[0] and array index 7 gets value[3].
+
+        Given the following array with a descending range:
+            Verilog [7:4]
+            VHDL    (7 downto 4)
+        Array index 7 gets value[0] and array index 4 gets value[3].
+
+        Given an HDL array, the values in the list will assigned to sub-handles
+        in the order indicated in the table:
+        +----------------+-----------------------+--------------------------------+
+        | Verilog        | VHDL                  | Array values                   |
+        +================+=======================+================================+
+        | ``array[4:7]`` | ``array(4 to 7)``     | ``array[4]`` gets ``value[0]`` |
+        |                |                       | ...                            |
+        |                |                       | ``array[7]`` gets ``value[3]`` |
+        +----------------+-----------------------+--------------------------------+
+        | ``array[7:4]`` | ``array(7 downto 4)`` | ``array[7]`` gets ``value[0]`` |
+        |                |                       | ...                            |
+        |                |                       | ``array[7]`` gets ``value[3]`` |
+        +----------------+-----------------------+--------------------------------+
+
+        """
```

Comment:
This docstring is not accessible unless you try really hard. You'll need to merge this with the getter docstring. I didn't notice it when I wrote my comment above